src/librustc_back/target/mod.rs

```diff
@@ -214,6 +214,7 @@ supported_targets! {
     ("le32-unknown-nacl", le32_unknown_nacl),
     ("asmjs-unknown-emscripten", asmjs_unknown_emscripten),
     ("wasm32-unknown-emscripten", wasm32_unknown_emscripten),
+    ("wasm32-unknown-unknown", wasm32_unknown_unknown),
```

Comment:
In any case I think this name is fine for now.

In the Rust system I would expect an emscripten-wasm target and a non-emscripten-wasm target to have two different target triples.